backend/src/schema/resolvers/moderation.js

```diff
@@ -1,41 +1,51 @@
+const transformReturnType = record => {
+  return {
+    ...record.get('review').properties,
+    report: record.get('report').properties,
+    resource: {
+      __typename: record.get('type'),
+      ...record.get('resource').properties,
+    },
+  }
+}
+
 export default {
   Mutation: {
-    disable: async (object, params, { user, driver }) => {
-      const { id } = params
-      const { id: userId } = user
-      const cypher = `
-      MATCH (u:User {id: $userId})
-      MATCH (resource {id: $id})
-      WHERE resource:User OR resource:Comment OR resource:Post
-      SET resource.disabled = true
-      MERGE (resource)<-[:DISABLED]-(u)
-      RETURN resource {.id}
-      `
-      const session = driver.session()
-      const res = await session.run(cypher, { id, userId })
-      session.close()
-      const [resource] = res.records.map(record => {
-        return record.get('resource')
-      })
-      if (!resource) return null
-      return resource.id
-    },
-    enable: async (object, params, { user, driver }) => {
-      const { id } = params
-      const cypher = `
-      MATCH (resource {id: $id})<-[d:DISABLED]-()
-      SET resource.disabled = false
-      DELETE d
-      RETURN resource {.id}
-      `
+    review: async (_object, params, context, _resolveInfo) => {
+      const { user: moderator, driver } = context
+
+      let createdRelationshipWithNestedAttributes = null // return value
       const session = driver.session()
-      const res = await session.run(cypher, { id })
-      session.close()
-      const [resource] = res.records.map(record => {
-        return record.get('resource')
-      })
-      if (!resource) return null
-      return resource.id
+      try {
+        const cypher = ` 
+            MATCH (moderator:User {id: $moderatorId})
+            MATCH (resource {id: $params.resourceId})<-[:BELONGS_TO]-(report:Report {closed: false})
+            WHERE resource:User OR resource:Post OR resource:Comment
+            MERGE (report)<-[review:REVIEWED]-(moderator)
+            ON CREATE SET review.createdAt = $dateTime, review.updatedAt = review.createdAt
+            ON MATCH SET review.updatedAt = $dateTime
+            SET review.disable = $params.disable, review.closed = $params.closed
```

Comment:
~Ah, so closing and reviewing can be the same action?~
After the reading below I ask myself, why do we need to store the `review.closed`? It's implicit because the most recent review must be the closing review.

Reply:
Actually, maybe we don't need... I think that @Tirokk was explicitly setting the `disable`/`closed` on the `REVIEWED` relationship when I starting refactoring. Maybe we can remove `closed` though since if the `Report` is `closed`, as you say it can only be the last `review` that closed it.